crates/ere-sp1/src/lib.rs

```diff
@@ -225,127 +225,62 @@ fn serialize_inputs(stdin: &mut SP1Stdin, inputs: &Input) {
-        let workspace_dir = env!("CARGO_WORKSPACE_DIR");
-        PathBuf::from(workspace_dir)
-            .join("tests")
-            .join("sp1")
-            .join("execute")
-            .join("basic")
-            .canonicalize()
-            .expect("Failed to find or canonicalize test guest program at <CARGO_WORKSPACE_DIR>/tests/execute/sp1")
+    fn basic_program() -> Vec<u8> {
+        BASIC_PRORGAM
+            .get_or_init(|| {
+                RV32_IM_SUCCINCT_ZKVM_ELF
+                    .compile(&testing_guest_directory("sp1", "basic"))
+                    .unwrap()
+            })
+            .to_vec()
     }
 
     #[test]
-    fn test_execute_sp1_dummy_input() {
-        let elf_bytes = get_compiled_test_sp1_elf()
-            .expect("Failed to compile test SP1 guest for execution test");
-
-        let mut input_builder = Input::new();
-        let n: u32 = 42;
-        let a: u16 = 42;
-        input_builder.write(n);
-        input_builder.write(a);
-
-        let zkvm = EreSP1::new(elf_bytes, ProverResourceType::Cpu);
+    fn test_execute() {
+        let program = basic_program();
+        let zkvm = EreSP1::new(program, ProverResourceType::Cpu);
+        let inputs = basic_inputs();
 
-        let result = zkvm.execute(&input_builder);
-
-        if let Err(err) = &result {
-            panic!("Execution error: {err}");
-        }
+        run_zkvm_execute(&zkvm, &inputs);
     }
 
     #[test]
-    fn test_execute_sp1_no_input_for_guest_expecting_input() {
-        let elf_bytes = get_compiled_test_sp1_elf()
-            .expect("Failed to compile test SP1 guest for execution test");
-
-        let empty_input = Input::new();
-
-        let zkvm = EreSP1::new(elf_bytes, ProverResourceType::Cpu);
-        let result = zkvm.execute(&empty_input);
-
-        assert!(
-            result.is_err(),
-            "execute should fail if guest expects input but none is provided."
-        );
-    }
-}
-
-#[cfg(test)]
-mod prove_tests {
-    use std::path::PathBuf;
-
-    use super::*;
-    use zkvm_interface::Input;
+    fn test_execute_empty_inputs() {
+        let program = basic_program();
+        let zkvm = EreSP1::new(program, ProverResourceType::Cpu);
+        let empty_inputs = Input::new();
 
-    fn get_prove_test_guest_program_path() -> PathBuf {
-        let workspace_dir = env!("CARGO_WORKSPACE_DIR");
-        PathBuf::from(workspace_dir)
-            .join("tests")
-            .join("sp1")
-            .join("prove")
-            .join("basic")
-            .canonicalize()
-            .expect("Failed to find or canonicalize test guest program at <CARGO_WORKSPACE_DIR>/tests/execute/sp1")
-    }
-
-    fn get_compiled_test_sp1_elf_for_prove() -> Result<Vec<u8>, SP1Error> {
-        let test_guest_path = get_prove_test_guest_program_path();
-        RV32_IM_SUCCINCT_ZKVM_ELF.compile(&test_guest_path)
+        run_zkvm_execute_invalid_inputs(&zkvm, &empty_inputs);
```

Comment:
Q: Maybe worth also some non-empty imput that isn't the expected `Hello world`?

Reply:
Q2/Nit: might be useful to remove `run_zkvm_execute_invalid_inputs` and only have the normal one and do the `.expect` here?

Reply:
> Q: Maybe worth also some non-empty imput that isn't the expected Hello world?

Sounds good

> Q2/Nit: might be useful to remove run_zkvm_execute_invalid_inputs and only have the normal one and do the .expect here?

Yeah we could do that, but then we will duplicate a lot of `.expect(msg)` in tests. Do you think we could just use `unwrap()`, or it's fine to duplicate the expect message?